### DISCUSSION TITLE
Error while using neuralcoref with spacy

### DISCUSSION THREAD

**Discussion by kausikb** [explosion/spaCy#13097] [Help: Coding & Implementations] [answered]
Here is the code I am using:
import spacy
import neuralcoref
nlp = spacy.load("en_core_web_sm")
neuralcoref.add_to_pipe(nlp)

When I run the above, I get the following error:
C:\Users\kausi\anaconda3\envs\py37\lib\importlib\_bootstrap.py:219: RuntimeWarning: spacy.morphology.Morphology size changed, may indicate binary incompatibility. Expected 104 from C header, got 112 from PyObject
  return f(*args, **kwds)
C:\Users\kausi\anaconda3\envs\py37\lib\importlib\_bootstrap.py:219: RuntimeWarning: spacy.vocab.Vocab size changed, may indicate binary incompatibility. Expected 96 from C header, got 112 from PyObject
  return f(*args, **kwds)
C:\Users\kausi\anaconda3\envs\py37\lib\importlib\_bootstrap.py:219: RuntimeWarning: spacy.tokens.span.Span size changed, may indicate binary incompatibility. Expected 72 from C header, got 80 from PyObject
  return f(*args, **kwds)

How do I fix this?

**Comment by kausikb** [comment]
Additional Details:

spaCy version    2.3.7
Location         C:\Users\kausi\anaconda3\envs\py37\lib\site-packages\spacy
Platform         Windows-10-10.0.22621-SP0
Python version   3.7.13
Models
Neuralcoref 4.0

**Comment by rmitsch** [comment]
Hi @kausikb, is transitioning to spaCy v3 an option for you?

**Reply by kausikb** [reply]
I guess transitioning to spaCy v3 is not an option, as I could figure out that Neuralcoref 4.0 works only with spaCy 2.1.0. 

Any new information on whether we can use the latest version of spaCy and Python will be very helpful.

**Reply by rmitsch** [reply]
You seem to be running this with v2.3.7 though. Does it work with v2.1.0?

**Reply by kausikb** [reply]
Neuralcoref 4.0 didn't work with spaCy 2.3.7. After a few iterations, when I switched to v2.1.0, it worked.

**Reply by rmitsch** [reply, marked as answer]
I'm afraid using NeuralCoref 4.0 only works with spaCy 2.1.0. I'd encourage you however to checkout spaCy's experimental [CorefResolver](https://spacy.io/api/coref) (see [here](https://explosion.ai/blog/coref) for more info and [here](https://github.com/explosion/projects/tree/v3/experimental/coref) for an example project) - this one works with new versions of spaCy.

**Reply by kausikb** [reply]
While using spaCy experimental package for CorefResolver, I am getting the following error, Any clue what package could be missing?

RegistryError: [E893] Could not find function 'spacy-experimental.Coref.v1' in function registry 'architectures'. If you're using a custom function, make sure the code is available. If the function is provided by a third-party package, e.g. spacy-transformers, make sure the package is installed in your environment.

Available names: spacy-legacy.CharacterEmbed.v1, spacy-legacy.EntityLinker.v1, spacy-legacy.HashEmbedCNN.v1, spacy-legacy.MaxoutWindowEncoder.v1, spacy-legacy.MishWindowEncoder.v1, spacy-legacy.MultiHashEmbed.v1, spacy-legacy.Tagger.v1, spacy-legacy.TextCatBOW.v1, spacy-legacy.TextCatCNN.v1, spacy-legacy.TextCatEnsemble.v1, spacy-legacy.Tok2Vec.v1, spacy-legacy.TransitionBasedParser.v1, spacy.CharacterEmbed.v2, spacy.EntityLinker.v2, spacy.HashEmbedCNN.v2, spacy.MaxoutWindowEncoder.v2, spacy.MishWindowEncoder.v2, spacy.MultiHashEmbed.v2, spacy.PretrainCharacters.v1, spacy.PretrainVectors.v1, spacy.SpanCategorizer.v1, spacy.Tagger.v2, spacy.TextCatBOW.v2, spacy.TextCatCNN.v2, spacy.TextCatEnsemble.v2, spacy.TextCatLowData.v1, spacy.Tok2Vec.v2, spacy.Tok2VecListener.v1, spacy.TorchBiLSTMEncoder.v1, spacy.TransitionBasedParser.v2

**Reply by rmitsch** [reply]
Please open a new issue for this. Thank you!
